### PR TITLE
chore/upgrade dependencies

### PR DESCRIPTION
**Pull request by sebbi08**

### CODE REVIEW

**Comment by jkoenig134** [comment]
you have to replace no-if with no-conditionals

File: .vscode/settings.json [hunk 1]
```diff
@@ -54,5 +54,6 @@
     "files.eol": "\n",
     "typescript.unstable": {
         "organizeImportsIgnoreCase": true
-    }
+    },
+    "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (jkoenig134) [inline .vscode/settings.json:58]: Why is this required?

Comment (sebbi08) [reply .vscode/settings.json:58]: VS Code is still shipped with ts 5.4 and this will use the locale ts version.

Comment (jkoenig134) [reply .vscode/settings.json:58]: according vscode docs it should ship with latest ts, but maybe it will change to 5.5 in the next version..

**Review by jkoenig134 — APPROVED** [review]

File: .ci/runChecks.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@ set -e
 
 npm ci
 npm run build:node
-npm run lint:eslint
+npm run lint:eslint -- --rule 'jest/no-conditional-in-test: off'
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply .ci/runChecks.sh:5]: if not ppl will take much time fixing things that doesn't have to be fixed

**Review by jkoenig134 — APPROVED** [review]
